## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#2127f879-32e8-44e5-a544-f48310d3203b) |
| mint-compression-dist-xl.sh | :x: [more...](#07c81a99-d74a-4318-8f03-fd01e21b7918) |
<a name="2127f879-32e8-44e5-a544-f48310d3203b"></a>
### 8418-2957486/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31939
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d606c07a9329:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 50 seconds
(4/14) Running aws-sdk-ruby tests ... done in 15 seconds
(5/14) Running awscli tests ... done in 1 minutes and 26 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 47 seconds
(8/14) Running minio-dotnet tests ... done in 2 minutes and 3 seconds
(9/14) Running minio-go tests ... FAILED in 5 minutes and 26 seconds
{
  "args": {},
  "duration": 118523,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="07c81a99-d74a-4318-8f03-fd01e21b7918"></a>
### 8418-2957486/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31291
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 24642901e207:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 2 minutes and 1 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 6 seconds
(8/14) Running minio-dotnet tests ... done in 49 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 17 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 40 seconds
(11/14) Running minio-js tests ... done in 55 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 23 seconds
{
  "status": "FAIL",
  "duration": 112,
  "message": "'content-language'",
  "name": "minio-py:test_presigned_get_object_response_headers",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1396, in test_presigned_get_object_response_headers\n    returned_content_language = response.headers['Content-Language']\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/_collections.py\", line 156, in __getitem__\n    val = self._container[key.lower()]\nKeyError: 'content-language'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2039, in main\n    test_presigned_get_object_response_headers(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1414, in test_presigned_get_object_response_headers\n    raise Exception(err)\nException: 'content-language'\n",
  "args": {
    "bucket_name": "minio-py-test-e85ad02a-a0cb-49f4-8b15-9ac731d41608",
    "object_name": "75ed59c0-2799-4636-ac99-4ddbcf2049ea",
    "presigned_get_object_url": "http://72.28.97.59:31291/minio-py-test-e85ad02a-a0cb-49f4-8b15-9ac731d41608/75ed59c0-2799-4636-ac99-4ddbcf2049ea?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191018%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191018T221120Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&response-content-language=en_US&response-content-type=text%2Fplain&X-Amz-Signature=8e3a2c4732fa9a8847a8d4987993fada7f56afa17f593ffc79ba22846d5c94f4"
  }
}

Executed 11 out of 14 tests successfully.
```
{"name": "8418-2957486", "full_size": 34544176, "images": [{"size": 34544176, "architecture": "amd64", "variant": null, "features": null, "os": "linux", "os_version": null, "os_features": null}], "id": 73112275, "repository": 757613, "creator": 21164, "last_updater": 21164, "last_updated": "2019-10-18T21:40:51.837338Z", "image_id": null, "v2": true}Error: No such image: minio/minio:8418-2957486

--- a/cmd/notification.go
+++ b/cmd/notification.go
@@ -1168,6 +1168,15 @@ func (args eventArgs) ToEvent() event.Event {
 func sendEvent(args eventArgs) {
 
 	// remove sensitive encryption entries in metadata.
+	switch {
+	case crypto.IsEncrypted(args.Object.UserDefined):
+		if totalObjectSize, err := args.Object.DecryptedSize(); err == nil {
+			args.Object.Size = totalObjectSize
+		}
+	case args.Object.IsCompressed():
+		args.Object.Size = args.Object.GetActualSize()
+	}
+
 	crypto.RemoveSensitiveEntries(args.Object.UserDefined)
 	crypto.RemoveInternalEntries(args.Object.UserDefined)
 

--- a/cmd/object-handlers.go
+++ b/cmd/object-handlers.go
@@ -1272,7 +1272,6 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 
 	if objectAPI.IsEncryptionSupported() {
 		if crypto.IsEncrypted(objInfo.UserDefined) {
-			objInfo.Size, _ = objInfo.DecryptedSize()
 			switch {
 			case crypto.S3.IsEncrypted(objInfo.UserDefined):
 				w.Header().Set(crypto.SSEHeader, crypto.SSEAlgorithmAES256)
@@ -2336,13 +2335,6 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 
 	// Write success response.
 	writeSuccessResponseXML(w, encodedSuccessResponse)
-
-	// Get host and port from Request.RemoteAddr.
-	if objectAPI.IsEncryptionSupported() {
-		if crypto.IsEncrypted(objInfo.UserDefined) {
-			objInfo.Size, _ = objInfo.DecryptedSize()
-		}
-	}
 
 	// Notify object created event.
 	sendEvent(eventArgs{

